src/java.desktop/share/classes/java/awt/MenuBar.java

```diff
@@ -339,15 +341,15 @@ final Menu getMenuImpl(int i) {
      * @since       1.1
      */
     public synchronized Enumeration<MenuShortcut> shortcuts() {
-        Vector<MenuShortcut> shortcuts = new Vector<>();
+        ArrayList<MenuShortcut> shortcuts = new ArrayList<>();
         int nmenus = getMenuCount();
         for (int i = 0 ; i < nmenus ; i++) {
             Enumeration<MenuShortcut> e = getMenu(i).shortcuts();
             while (e.hasMoreElements()) {
-                shortcuts.addElement(e.nextElement());
+                shortcuts.add(e.nextElement());
             }
         }
-        return shortcuts.elements();
```

Comment:
Can you please confirm the old and new enumerations have the same behavior during iteration/adding/removing/etc elements?

Reply:
Behavior is the same.
`Enumeration` doesn't have `remove` method. Only `hasMoreElements` and `nextElement`.
Method creates a new collection for result each time, hence no updates of underlying collection are possible.
Iteration order is the same too.
```
        Vector<String> v = new Vector<>();
        v.add("123");
        v.add("444");
        v.add("555");
        v.add("666");

        Enumeration<String> elements = v.elements();
        while (elements.hasMoreElements()) {
            String s = elements.nextElement();
            System.out.println(s);
        }
        System.out.println();

        ArrayList<String> a = new ArrayList<>(v);
        Enumeration<String> e = Collections.enumeration(a);
        while (e.hasMoreElements()) {
            String s = e.nextElement();
            System.out.println(s);
        }
```
gives
```
123
444
555
666

123
444
555
666
```

Reply:
but there is an asIterator method that may support remove